spec/controllers/api/students_controller_spec.rb

```diff
@@ -989,4 +989,83 @@
     end
   end
 
+  describe 'GET inactive' do
+    context 'when the user is signed in' do
+      let(:user) { FactoryBot.create(:user) }
+
+      subject do
+        request.headers['Authorization'] = "Bearer #{generate_token(user)}"
+        get :index, params: { format: :json }
+
+        response
+      end
+
+      context 'with students' do
+        let(:student) { FactoryBot.create(:student) }
```

Comment:
Esto podés usar los traits que mapea factory bot a enums de una, es decir, esto funciona out of the box `FactoryBot.create(:student, :inactive)`

Reply:
Así te evitás el bloque before de abajo